### PR TITLE
Feature flagging

### PR DESCRIPTION
**Pull request by roncohen**

### CODE REVIEW

File: src/flags.ts [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+      }
+    } else {
+      result[key] = obj[key];
+    }
+  }
+  return result;
+}
+
+const localStorageCacheKey = `__bucket_flags`;
+interface cacheEntry {
```
Comment (rrpff) [inline src/flags.ts:133]: This cache stuff is fairly dense, don't think it'd hurt to make it more generic (looks it already, could just use a key other than `flags`) and split out into a separate file with some unit tests

**Review by rrpff — APPROVED** [review]
Looks solid 👍

File: src/main.ts [hunk 2]
```diff
@@ -558,6 +562,64 @@ export default function main() {
...
+   *                argument.
+   * @param fallbackFlags Optional array of flags to use if the request to Bucket fails.
+   * @param includeFlags Optional array of flags to include in the response automatically.
+   * @param timeoutMs Optional the timeout in milliseconds for the request to Bucket before using `fallbackFlags`. Default is 5000ms.
+   * @param staleWhileRevalidate Optional whether to return potentially stale flags while revalidating the flags in the background. Default is true.
+   */
+  async function getFeatureFlags({
+    context,
+    fallbackFlags = [],
+    includeFlags = [],
```
[the remark this answers is not in the document]
Comment (roncohen) [reply src/main.ts:583]: good point, I'll just take it out.

File: src/flags.ts (hunk 1 above)
Comment (roncohen) [reply src/flags.ts:133]: great idea!

**Pull request merged**